services/consensusalgo/leanhelixconsensus/test/metrics_test.go

```diff
@@ -0,0 +1,144 @@
+// +build !race
+
+package test
+
+import (
+	"context"
+	"github.com/orbs-network/lean-helix-go/services/blockproof"
+	"github.com/orbs-network/lean-helix-go/services/interfaces"
+	"github.com/orbs-network/lean-helix-go/services/randomseed"
+	"github.com/orbs-network/lean-helix-go/spec/types/go/protocol"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/crypto/keys"
+	"github.com/orbs-network/orbs-network-go/test/with"
+	"github.com/stretchr/testify/require"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestMetricsAreUpdatedOnElectionTrigger(t *testing.T) {
+
+	with.Concurrency(t, func(ctx context.Context, parent *with.ConcurrencyHarness) {
+		h := newSingleLhcNodeHarness().
+			withBaseConsensusRoundTimeout(10*time.Millisecond).
+			start(parent, ctx)
+
+		h.beFirstInCommittee() // just so RequestOrderedCommittee will succeed
+		h.expectGossipSendLeanHelixMessage()
+
+		h.handleBlockSync(ctx, 1)
+
+		// Election trigger should fire and metrics should be updated
+
+		metrics := h.getMetrics()
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.currentElectionCount.Value() > 0
+		}), "expected currentElectionCount metric to update")
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.currentLeaderMemberId.Value() != ""
+		}), "expected currentLeaderMemberId metric to update")
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			matched, err := regexp.MatchString("samples=[1-9]", metrics.timeSinceLastElectionMillis.String())
+			if err != nil {
+				panic(err)
+			}
+			return matched
+		}), "expected timeSinceLastElectionMillis metric to update")
+	})
+
+}
+
+func TestMetricsAreUpdatedOnCommit(t *testing.T) {
+
+	with.Concurrency(t, func(ctx context.Context, parent *with.ConcurrencyHarness) {
+		h := newSingleLhcNodeHarness().start(parent, ctx)
+
+		h.expectGossipSendLeanHelixMessage()
+		h.expectValidateTransactionBlock()
+		h.expectValidateResultsBlock()
+		h.expectCommitBlock()
+
+		h.dontBeFirstInCommitee()
+
+		const syncedBlockHeight = 1
+		h.handleBlockSync(ctx, syncedBlockHeight)
+
+		orderedCommittee := h.requestOrderingCommittee(ctx)
+		orderedComitteeNodeIndicies := keys.NodeAddressesForTestsToIndexes(orderedCommittee.NodeAddresses)
+
+		leaderNodeIndex := orderedComitteeNodeIndicies[0] // leader for view 0 is always the first member in the ordered committee
+
+		prevBlockProof := protocol.BlockProofReader(nil) // the previous block (from sync) had no proof
```

Comment:
this is noise, doesn't pertain to the test itself. Can you extract block generation to a named function or to the harness?